microsoft/azuredatastudio
# ADS Enhancement - Support Sending Query Results to a File

This PR fixes #86

### Description
The change set in this PR is the first iteration to enhance ADS to support sending query results to a file. To enable sending query results to a file, open the command palette and enter "Query Results to File".
![image](https://user-images.githubusercontent.com/87730006/146491976-e7bc1732-a99a-4f3c-a77b-e331ffc9ff6e.png)

As can be seen in the screen shot, the "Query Results to Grid" option allows the user to go back to sending query results to a grid, if they are wanting to no longer send results to a file.

### Testing
_1. Query Results to File:_
- Launch ADS
- Open the command palette, and select "Query Results to File"
- Execute any query
- ADS will open an untitled editor that includes both the query results and messages and doesn't mark the editor as needing to be saved for quick closing.

When query results are sent to a file, results appear like this:
![image](https://user-images.githubusercontent.com/87730006/148109542-672c89c3-720b-47f6-9f34-5e50590d9cc6.png)




_2. Query Results to Grid_
- Launch ADS
- Open the command palette, and select "Query Results to Grid"
- Execute any query
- ADS will send the query results to the grid in the "Results" tab and messages will appear in the "Messages" tab like they have been appearing in ADS.

When query results are sent to a grid, results appear like this:
![image](https://user-images.githubusercontent.com/87730006/148109657-95653778-adc8-4086-b4f2-2d47d73e6b7e.png)
